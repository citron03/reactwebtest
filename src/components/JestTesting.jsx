export function plus(a, b) {
  return a + b;
}

const JestTesting = () => {
  return (
    <div>
      <h1>H1</h1>
      <div>
        <p>text</p>
        <button onClick={plus(-1, -2)} placeholder="placeholder">
          button
        </button>
      </div>
    </div>
  );
};

export default JestTesting;
