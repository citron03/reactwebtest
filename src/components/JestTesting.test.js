import renderer from "react-test-renderer";
import { plus } from "./JestTesting";

test("plus function", () => {
  expect(plus(1, 2)).toBe(3);
});

it("click button", () => {
  const component = renderer.create(<button>button</button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onClick = plus(-1, -2);
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
