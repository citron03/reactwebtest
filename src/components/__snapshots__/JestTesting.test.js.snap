// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<button>
  button
</button>
`;

exports[`click button 1`] = `
<button>
  button
</button>
`;

exports[`click button 2`] = `
<button>
  button
</button>
`;
